components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -1034,7 +1034,14 @@ public boolean testRead(String[] args)
 
     // initialize reader
     long s = System.currentTimeMillis();
-    reader.setId(id);
+    try {
+      reader.setId(id);
+    } catch (FormatException exc) {
+      reader.close();
```

Comment:
Agreed. The most appropriate proposal I can come with might be to migrate this `try/catch` behavior into `FormatReader.setId(String)`. As there might be caveats we want to discuss with /cc @melissalinkert , I propose to push this change out of the PR and start by fixing the NPE.